### PR TITLE
Ab/user legal address

### PR DESCRIPTION
**Pull request by abeglova**
## Description
This PR imports the user_legaladdress table into staging

## Motivation and Context
https://github.com/mitodl/ol-data-platform/issues/336

## How Has This Been Tested?
This has been tested locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (improves on existing behavior)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project. (Did you install and run the pre-commit hooks?)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/ol_dbt/models/staging/mitxonline/_stg_mitxonline__models.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    - not_null
+  - name: user_address_country
+    description: string, user country code
+  - name: user_id
+    description: int, foreign key to users_user
+    tests:
+    - unique
+    - not_null
+  - name: last_name
+    description: string, user last name
```
Comment (rachellougee) [inline src/ol_dbt/models/staging/mitxonline/_stg_mitxonline__models.yml:20]: Should we test not_null for user_address_country, last_name, and first_name? they are not null in the source tables

**Review by rachellougee — APPROVED** [review]
Looks good, Just a comment for not null fields.

File: src/ol_dbt/models/staging/mitxonline/_stg_mitxonline__models.yml (hunk 1 above)
Comment (abeglova) [reply src/ol_dbt/models/staging/mitxonline/_stg_mitxonline__models.yml:20]: I'll add not null tests for this

Comment (dseaton) [reply src/ol_dbt/models/staging/mitxonline/_stg_mitxonline__models.yml:20]: @pdpinch Can I create an account in MITx Online without specifying a full name?

**Review by dseaton — APPROVED** [review]
One comment on tests. My original discussion with @abeglova involved testing that ensured certain fields were usable in intermediate and mart models. @rachellougee suggested some tests (not null) based on the pre-transformed state of the data, which wasn't on my radar for the staging layer. 

I approve this PR, and suggest, we find a venue to talk about testing strategies for dbt.

File: src/ol_dbt/models/staging/mitxonline/_stg_mitxonline__models.yml (hunk 1 above)
Comment (abeglova) [reply src/ol_dbt/models/staging/mitxonline/_stg_mitxonline__models.yml:20]: There's a non-null constraint on the name (full name) in the postges database

Comment (abeglova) [reply src/ol_dbt/models/staging/mitxonline/_stg_mitxonline__models.yml:20]: So you can't save a new user without a name

Comment (abeglova) [reply src/ol_dbt/models/staging/mitxonline/_stg_mitxonline__models.yml:20]: Same for the fields Rachel pointed out in user_legaladdress

**Pull request merged**